heron/executor/src/python/heron_executor.py

```diff
@@ -147,6 +148,7 @@ def __init__(self, args, shell_env):
         base64.b64decode(parsed_args.instance_jvm_opts.lstrip('"').
                          rstrip('"').replace('&equals;', '='))
     self.classpath = parsed_args.classpath
+    self.master_host = os.environ.get('HOST') if 'HOST' in os.environ else socket.gethostname()
```

Comment:
I need to go and verify this -- not sure if there's another way to get the actual hostname but i'll see what i can find